packages/lambda-xyz/src/tile.set.cache.ts

```diff
@@ -9,20 +9,104 @@ export function getTileSet(name: string, projection: Epsg): TileSet | undefined
     return TileSets.get(tileSetId);
 }
 
-export async function loadTileSet(req: LambdaContext, name: string, projection: Epsg): Promise<TileSet | null> {
-    req.set('tileSet', name);
-    req.set('projection', projection);
+export async function loadTileSet(name: string, projection: Epsg): Promise<TileSet | null> {
     let tileSet = getTileSet(name, projection);
     if (tileSet == null) {
         tileSet = new TileSet(name, projection);
         TileSets.set(tileSet.id, tileSet);
     }
-    req.timer.start('tileset:load');
     const loaded = await tileSet.load();
-    req.timer.end('tileset:load');
     if (!loaded) {
         TileSets.delete(tileSet.id);
         return null;
     }
     return tileSet;
 }
+
+function titleizeName(name: string): string {
+    return name.replace(/_0-/g, ' 0.').replace(/_/g, ' ');
+}
+
+function individualTileSet(parent: TileSet, image: TileSetRuleImagery): TileSet {
+    const id = image.imagery.id.slice(3);
+    const copy = new TileSet(id, parent.projection);
+    // use parent data as prototype for child;
+    copy.tileSet = Object.create(parent.tileSet ?? null);
+
+    const { tileSet } = copy;
+
+    copy.titleOverride = `${parent.title} ${titleizeName(image.imagery.name)}`;
+    copy.extentOverride = Bounds.fromJson(image.imagery.bounds).toBbox();
+
+    const rule = {
+        id,
+        minZoom: 0,
+        maxZoom: image.rule.maxZoom,
+        priority: 0,
+    };
+    // Override imagery (for completeness)
+    tileSet.imagery = { [id]: rule };
+    copy.tileSet = tileSet;
+
+    copy.imagery = [
+        {
+            rule,
+            imagery: image.imagery,
+        },
+    ];
+
+    return copy;
+}
+
+async function collectTileSets(tileSets: TileSet[], name: string, projection: Epsg | null): Promise<void> {
+    if (projection == null) {
+        for (const code of ProjectionTileMatrixSet.targetCodes()) {
+            const epsg = Epsg.get(code);
+            await collectTileSets(tileSets, name, epsg);
+        }
+        return;
+    }
+
+    const parent = await loadTileSet(name, projection);
+    if (parent != null) {
+        tileSets.push(parent);
+        if (parent.imagery != null) {
+            for (const image of parent.imagery) {
+                tileSets.push(individualTileSet(parent, image));
+            }
+        }
+    }
+}
+
+function compareByTitle(a: TileSet, b: TileSet): number {
+    return a.title.localeCompare(b.title);
+}
+
+/**
+ * Load a collection of TileSets. Used by WMTSCapabilities
+
+ * @param name if an empty string load all TileSets
+ * @param projection if null load all projections
+ */
+export async function loadTileSets(nameStr: string, projection: Epsg | null): Promise<TileSet[]> {
+    if (nameStr !== '' && nameStr[0] !== '@' && projection != null) {
```

Comment:
why can a nameStr startwith '@' ?

Reply:
Do that you can request all layers for a tagged tileSet for example: `http://localhost:5050/v1/tiles/@head/WMTSCapabilities.xml`